### ISSUE TITLE
[vscode.typescript-language-features] Comment/Uncomment does not work with spaced JSX comments

### ISSUE THREAD

**Issue by HughxDev** [microsoft/vscode#131122]
Does this issue occur when all extensions are disabled?: Yes

- VS Code Version:
  - 1.59.0
  - 1.60.0-insider (Universal)
- OS Version: macOS Big Sur 11.4 (20F71)

Steps to Reproduce:

1. Comment out a run of JSX code with spaces between the curly brackets, e.g. `{ /* { data } */ }`.
2. Try to uncomment it using VS Code shortcut <kbd>Cmd</kbd> + <kbd>/</kbd>

Expected result: Uncommented code, `{ data }`
Actual result: Double-commented code, `{/* { /* { data } */ } */}`

This is particularly frustrating if you have an ESLint plugin set up to auto-format your code, and you are using a rule like [`react/jsx-curly-spacing`](https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/jsx-curly-spacing.md), which converts all `foo={bar}` to `foo={ bar }` and and `{bar}` to `{ bar }`. So you will wind up doing <kbd>Cmd</kbd> + <kbd>/</kbd> to comment something out, save or otherwise trigger the ESLint auto-fix which adds spaces, then go to do <kbd>Cmd</kbd> + <kbd>/</kbd> again which no longer uncomments as expected.

**Comment by mjbvz** [comment]
This seems to work for me. Please share example code with extract steps to reproduce

**Comment by HughxDev** [comment]
@mjbvz Sorry, I guess I missed a step. Looks like this kind of comment is only activated for nested JSX tags. Here is a video reproduction:

https://user-images.githubusercontent.com/229232/129975052-73b80519-b23a-4183-a03a-5eb65cb4e966.mp4

And here is example code:

https://gist.github.com/HughxDev/c03ee71f39a31c044895193a8bf348c2

**Comment by mjbvz** [comment]
Thanks

Commenting and then uncommenting with cmd+/ worlds fine in my testing. The reason it seems to be failing here is the spaces between `{` and the `/*`


Depends on https://github.com/microsoft/vscode/issues/64659

**Comment by HughxDev** [comment]
@mjbvz 

> Commenting and then uncommenting with cmd+/ worlds fine in my testing. The reason it seems to be failing here is the spaces between `{` and the `/*`

Yes, that’s the basis for this issue. Note the title says “spaced JSX comments”. I want to be able to Cmd + / regardless of whether or not there are spaces between the curly brackets, because my linter is set up to automatically add spaces between all JSX curly brackets.

**Comment by HughxDev** [comment]
I believe this should be categorized as a bug and not a feature request, as the JSX format allows arbitrary spaces within curly brackets. So comment toggling should cater to this inherent flexibility as part of spec compliance, rather than impose an arbitrary restriction on how an author chooses to write comments.

**Comment by VSCodeTriageBot** [comment]
We closed this issue because we [don't plan to address it](https://aka.ms/vscode-out-of-scope) in the foreseeable future. If you disagree and feel that this issue is crucial: we are happy to listen and to reconsider.

If you wonder what we are up to, please see our [roadmap](https://aka.ms/vscoderoadmap) and [issue reporting guidelines](https://aka.ms/vscodeissuereporting).

Thanks for your understanding, and happy coding!

**Comment by HughxDev** [comment]
This should not be closed as this is not an “out-of-scope feature request” but rather a JSX spec compliance bug. Plenty of developers use spacing to increase the readability of code.

**Comment by mjbvz** [comment]
@HughxDev  This issue has gotten no upvotes in over a year so we have no plans to prioritize it. It's also blocked by #64659
